src/filepath.c

```diff
@@ -3136,8 +3210,14 @@ concat_fnames(char_u *fname1, char_u *fname2, int sep)
     void
 add_pathsep(char_u *p)
 {
-    if (*p != NUL && !after_pathsep(p, p + STRLEN(p)))
-	STRCAT(p, PATHSEPSTR);
+    size_t  plen;
+
+    if (p == NULL || *p == NUL)
+	return;
+
+    plen = STRLEN(p);
+    if (!after_pathsep(p, p + plen))
+	STRCPY(p + plen, PATHSEPSTR);
```

Comment:
This does not seem to be a clear advantage compared to:
```suggestion
	STRCPY(p + STRLEN(p), PATHSEPSTR);
```

Reply:
`plen` is also used in the condition above

Reply:
that is true, but essentially, what the patch does is only check for `p != NULL`. So in other words, this would have been equal, no?
```c
  add_pathsep(char_u *p)
  {
      if (p != NULL && *p != NUL && !after_pathsep(p, p + STRLEN(p)))
          STRCAT(p, PATHSEPSTR);
  }
```

Reply:
Not exactly, because STRCAT() calls STRLEN() as well. By calling STRLEN(p), storing the result in plen and using that means we only call STRLEN() once, not twice.